design/yugabyte/yugabytedb-rook-design.md

```diff
@@ -0,0 +1,89 @@
+metadata:
+  name: rook-yugabytedb
+  namespace: rook-yugabytedb
+spec:
+  master:
+    # Replica count for Master.
+    replicas: 3
+    # Mentioning network ports is optional. If some or all ports are not specified, then they will be defaulted to below-mentioned values, except for tserver-ui.
+    network:
+      ports:
+        - name: yb-master-ui
+          port: 7000          # default value
+        - name: yb-master-grpc
+          port: 7100          # default value
+    # Volume claim template for Master
+    volumeClaimTemplate:
+      metadata:
+        name: datadir
+      spec:
+        accessModes: [ "ReadWriteOnce" ]
+        resources:
+          requests:
+            storage: 1Gi
+        storageClassName: standard
+  tserver:
+    # Replica count for TServer
+    replicas: 3
+    # Mentioning network ports is optional. If some or all ports are not specified, then they will be defaulted to below-mentioned values, except for tserver-ui.
+    # For tserver-ui a cluster ip service will be created if the yb-tserver-ui port is explicitly mentioned. If it is not specified, only StatefulSet & headless service will be created for TServer. TServer ClusterIP service creation will be skipped. Whereas for Master, all 3 kubernetes objects will always be created.
+    network:
+      ports:
+        - name: yb-tserver-ui
+          port: 9000
+        - name: yb-tserver-grpc
+          port: 9100          # default value
+        - name: yb-tserver-cassandra
+          port: 9042          # default value
+        - name: yb-tserver-redis
+          port: 6379          # default value
+        - name: yb-tserver-postgres
+          port: 5433          # default value
+    # Volume claim template for TServer
+    volumeClaimTemplate:
+      metadata:
+        name: datadir
+      spec:
+        accessModes: [ "ReadWriteOnce" ]
+        resources:
+          requests:
+            storage: 1Gi
+        storageClassName: standard
+```
+# Cluster Settings
+
+### Master/TServer
+Master & TServer are two essential components of a YugabyteDB cluster. Specify Master/TServer specific attributes under `master`/`tserver`. The valid attributes are `replicas`, `network` & `volumeClaimTemplates`.
+
+### Replica Count
+Specify replica count for `master` & `tserver` pods under `replicas` field. This is a **required** field.
+
+### Network
+`network` field accepts `NetworkSpec` to be specified which describes YugabyteDB network settings. This is an **optional** field. Default network settings will be used, if any or all of the acceptable values are absent.
+
+A ClusterIP service will be created when `yb-tserver-ui` port is explicitly specified. If it is not specified, only StatefulSet & headless service will be created for TServer. ClusterIP service creation will be skipped. Whereas for Master, all 3 kubernetes objects will always be created.
+
+The acceptable port names & their default values are as follows:
+
+| Name | Default Value |
```

Comment:
should the `yb-tserver-ui` port be listed in this table too?

Reply:
Yb-tserver-ui port has to be explicitly specified if you want the UI pod. Hence not included in default port table